website/addons/wiki/views.py

```diff
@@ -218,6 +218,9 @@ def project_wiki_delete(auth, wname, **kwargs):
 
     if not wiki_page:
         raise HTTPError(http.NOT_FOUND)
+    comments = getattr(node.get_wiki_page(wiki_name, 1), 'commented', [])
+    for comment in comments:
```

Comment:
I think hiding comments should probably happen after `node.delete_node_wiki` is called. We only want to hide the comments if `node.delete_node_wiki`  completes successfully.

Also, since hiding of comments should *always* happen when `delete_node_wiki` is invoked, it might be better to use a signal-subscriber (using Blinker) to make sure that comments are hidden.

For an example usage of signals, see the User model: https://github.com/huangginny/osf.io/blob/ab7ae19ea8145b143621dadef30aaab529baabde/framework/auth/core.py#L407-408